addon/models/node.js

```diff
@@ -139,5 +139,87 @@ export default OsfModel.extend(FileItemMixin, {
         );
         this.set('_dirtyRelationships.contributors.remove', []);
         return promise;
-    }
+    },
+
+    addChild(title, description, category) {
+        let child = this.store.createRecord('node', {
+            title: title,
+            category: category || 'project',
+            description: description || null
+        });
+
+        return this.store.adapterFor('node').ajax(this.get('links.relationships.children.links.related.href'), 'POST', {
+            data: child.serialize(),
+        }).then(resp => {
+            child.unloadRecord();
+            this.store.pushPayload(resp);
+            let created = this.store.peekRecord('node', resp.data.id);
+            this.get('children').pushObject(created);
+            return created;
+        });
+    },
+
+    addUnregisteredContributor(fullName, email, permission, isBibliographic, index=Number.MAX_SAFE_INTEGER) {
+        let user = this.store.createRecord('user', {
+            fullName: fullName,
+            username: email
+        });
+        // After user has been saved, add user as a contributor
+        return user.save().then(user => this.addContributor(user.id, permission, isBibliographic, index));
+
+    },
+
+    addContributor(userId, permission, isBibliographic, index=Number.MAX_SAFE_INTEGER) {
+        let contrib = this.store.createRecord('contributor', {
+            // Original code used the line below.
+            // Serialize does something weird I guess
+            // id: `${this.get('id')}-${userId}`,
+            id: userId,
```

Comment:
Before when contributors were being created, it ran through osf-adapter/create-related, and we added 'create' to adapterOptions here https://github.com/CenterForOpenScience/ember-osf/blob/develop/addon/adapters/osf-adapter.js#L81-L84.

Then when it got to the contributor serializer, we would see the 'create' attached in previous step and know to use userId instead of contributor id. https://github.com/CenterForOpenScience/ember-osf/blob/develop/addon/serializers/contributor.js#L9-L10

The same contributor serializer is handling both create and update requests, but create method needs user id and update method needs contributor id.